nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-processors/src/main/java/org/apache/nifi/processors/elasticsearch/AbstractElasticsearchHttpProcessor.java

```diff
@@ -0,0 +1,161 @@
+import java.net.Proxy;
+import java.net.URL;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * A base class for Elasticsearch processors that use the HTTP API
+ */
+public abstract class AbstractElasticsearchHttpProcessor extends AbstractElasticsearchProcessor {
+
+    public static final PropertyDescriptor ES_URL = new PropertyDescriptor.Builder()
+            .name("elasticsearch-http-url")
+            .displayName("Elasticsearch URL")
+            .description("Elasticsearch URL which will be connected to, including scheme, host, port, path. The default port for the REST API is 9200.")
+            .required(true)
+            .addValidator(StandardValidators.URL_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor PROXY_HOST = new PropertyDescriptor.Builder()
+            .name("elasticsearch-http-proxy-host")
+            .displayName("Proxy Host")
+            .description("The fully qualified hostname or IP address of the proxy server")
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor PROXY_PORT = new PropertyDescriptor.Builder()
+            .name("elasticsearch-http-proxy-port")
+            .displayName("Proxy Port")
+            .description("The port of the proxy server")
+            .required(false)
+            .addValidator(StandardValidators.PORT_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor CONNECT_TIMEOUT = new PropertyDescriptor.Builder()
+            .name("elasticsearch-http-connect-timeout")
+            .displayName("Connection Timeout")
+            .description("Max wait time for the connection to the Elasticsearch REST API.")
+            .required(true)
+            .defaultValue("5 secs")
+            .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor RESPONSE_TIMEOUT = new PropertyDescriptor.Builder()
+            .name("elasticsearch-http-response-timeout")
+            .displayName("Response Timeout")
+            .description("Max wait time for a response from the Elasticsearch REST API.")
+            .required(true)
+            .defaultValue("15 secs")
+            .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
+            .build();
+
+    private final AtomicReference<OkHttpClient> okHttpClientAtomicReference = new AtomicReference<>();
+
+    @Override
+    protected void createElasticsearchClient(ProcessContext context) throws ProcessException {
+        okHttpClientAtomicReference.set(null);
+
+        OkHttpClient okHttpClient = new OkHttpClient();
+
+        // Add a proxy if set
+        final String proxyHost = context.getProperty(PROXY_HOST).getValue();
+        final Integer proxyPort = context.getProperty(PROXY_PORT).asInteger();
+        if (proxyHost != null && proxyPort != null) {
```

Comment:
May want to add a custom validator to check if one is set the other has to be set.